deepspeed/runtime/zero/stage2.py

```diff
@@ -775,8 +778,11 @@ def get_grad_position(self, group_id, tensor_list, first_offset, partition_size)
             current_offset += num_elements
 
     def update_overflow_tracker_for_param_grad(self, param):
-        if param.grad is not None and self._has_inf_or_nan(param.grad.data):
-            self.local_overflow = True
+        if param.grad is not None:
+            if self.overlap_comm:
+                self.gpu_sum = self.gpu_sum + param.grad.data.float().sum()
+            elif self._has_inf_or_nan(param.grad.data):
+                self.overflow = True
```

Comment:
This should be self.local_overflow not self.overflow?

Reply:
Right, I fixed this.